### PR TITLE
Find references to a function outside module

### LINKED ISSUE #4188: Finding references/callers of associated functions doesn't work across module boundaries
```rust
mod foo {
    pub struct Foo;

    impl Foo {
        pub fn new() -> Foo {
            Foo
        }
    }
}

fn main() {
    let _f = foo::Foo::new();
}
```

"Go to definition" on the `new()` call inside `main()` correctly jumps to the definition of the function. From there finding references (i.e. callers) doesn't yield any results though.

When removing the `foo` module it finds the caller correctly. Similarly the same works fine for non-associated, free-standing functions inside the `foo` module.

### PR DESCRIPTION
**Pull request by montekki**
Fixes #4188 

Yet again, it looks like although the code in 
https://github.com/rust-analyzer/rust-analyzer/blob/da1f316b0246ce41d7cb8560181e294089f06ef3/crates/ra_ide_db/src/search.rs#L128-L132

may be wrong, it is not hit since the `vis` is `None` at this point. The fix is similar to the #4237 case: just add another special case to `Definition::visibility()`.

### CODE REVIEW

File: crates/ra_ide_db/src/defs.rs [hunk 1]
```diff
@@ -52,6 +52,7 @@ impl Definition {
                     let parent = id.parent_enum(db);
                     module?.visibility_of(db, &ModuleDef::Adt(Adt::Enum(parent)))
                 }
+                ModuleDef::Function(f) => Some(f.visibility(db)),
```
Comment (flodiebold) [inline crates/ra_ide_db/src/defs.rs:55]: we should do the same for consts and type aliases.

Comment (flodiebold) [reply crates/ra_ide_db/src/defs.rs:55]: Also, maybe we should move this logic into a function in `code_model` called something like `definition_visibility` :thinking:

Comment (montekki) [reply crates/ra_ide_db/src/defs.rs]: done it in two separate commits

**Pull request merged**